homeassistant/components/google_assistant/trait.py

```diff
@@ -1047,3 +1036,67 @@ def query_attributes(self):
                             ATTR_ENTITY_ID: self.state.entity_id,
                             media_player.ATTR_INPUT_SOURCE: source
                         }, blocking=True, context=data.context)
+
+
+@register_trait
+class OpenCloseTrait(_Trait):
+    """Trait to open and close a cover.
+    https://developers.google.com/actions/smarthome/traits/openclose
+    """
+
+    name = TRAIT_OPENCLOSE
+    commands = [
+        COMMAND_OPENCLOSE
+    ]
+
+    @staticmethod
+    def supported(domain, features):
+        """Test if state is supported."""
+        return domain == cover.DOMAIN
+
+    def sync_attributes(self):
+        """Return opening direction."""
+        return {
+            'openDirection': ['UP']
+        }
+
+    def query_attributes(self):
+        """Return state query attributes."""
+        domain = self.state.domain
+        response = {}
+
+        if domain == cover.DOMAIN:
+            position = self.state.attributes.get(cover.ATTR_CURRENT_POSITION)
+            if position is not None:
+                response['openPercent'] = position
+                response['on'] = self.state.state != cover.STATE_CLOSED
+                response['online'] = True
```

Comment:
HA should be sending the online value on its own I believe

Reply:
I also couldn't find anywhere that mentioned if the 'on' trait actually does what you are thinking here. I think this is more for devices that need to be woken up before they do anything and not for the open/close state.

Reply:
We always add online upstream in our code, this line is not necessary.